docs/components/Date.md

```diff
@@ -0,0 +1,88 @@
+- **Year input is less than 4 digits in length** - Triggers when the year field is visible and the user enters a year input which is less than 4 digits in length. For example, the inputs "Day: 31, Month: 3, Year: 165" and "Month: 2, Year: 12".
+- **Any of the inputs that are visible are not an integer** - Triggers when the user enters a numerical value that is not an integer into any of the day, month or year inputs when they are visible. For example, the input "Day: -1, Month: 4, Year: -5".
+- **Month input is less than 1 or greater than 12** - Triggers when the user enters a month input which is less than 1 or greater than 12. For example, the inputs "Day: 24, Month: 0, Year: 2021" and "Day: 12, Month: 15, Year: 2008".
+
+### Relational dates
+When using the minimum and maximum date input properties (MinDate and MaxDate respectively), the below list of input cases will also trigger an error.
+
+- **Date is in the future when it needs to be the current day or in the past** - Triggers when the minimum date set in the MinDate input property is the default, the maximum date set in the MaxDate input property is set as the current day and the date entered by the user is after the current day. For example, if the current day is 19th August 2024, the input "Day: 21, Month: 8, Year: 2024" will trigger an error.
+- **Date is in the past when it needs to be the current day or in the future** - Triggers when the minimum date set in the MinDate input property is set as the current day, the maximum date in the MaxDate input property is the default and the date entered by the user is before the current day. For example, if the current day is 14th July 2024, the input "Day: 10, Month: 6, Year: 2023" will trigger an error.
+- **Date is not the same as or after the MinDate** - Triggers when the minimum date has been changed from the default in the MinDate input property, the maximum date in the MaxDate input property is the default and the date entered by the user is before the minimum date. For example, if the MinDate is set as 28th February 2024, the input "Day: 24, Month: 2, Year: 2024" will trigger an error.
+- **Date is not the same as or before the MaxDate** - Triggers when the minimum date in the MinDate input property is the default, the maximum date has been changed from the default in the MaxDate input property and the date entered by the user is after the maximum date. For example, if the the MaxDate is set as 19th November 2005, the input "Day: 25, Month: 1, Year: 2006" will trigger an error.
+- **Date is not between the MinDate and the MaxDate (inclusive)** - Triggers when the date input is before the minimum date set in the MinDate input property or after the maximum date set in the MaxDate input property. For example, if MinDate is DateValue("13/01/2024") and MaxDate is DateValue("20/01/2024"), the inputs "Day: 11, Month: 1, Year: 2024" and "Day: 22, Month: 1, Year: 2024" will trigger an error.
```

Comment:
We could mention order of errors triggering, we have the order "missing fields, invalid fields, relational errors" as per GDS